### PR TITLE
Fix data copy method in state() 

### PR DESCRIPTION
**Pull request by multiphaseCFD**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
Fix the bug in state() method. 
**Description of the Change:**
The state is now copied to host in state() method.
**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -67,6 +67,8 @@ and change `lightning.gpu` to inherit from `QubitDevice` instead of `LightningQu
 * Pin CMake to 3.24.x in wheel-builder to avoid Python not found error in CMake 3.25.
 [(#75)](https://github.com/PennyLaneAI/pennylane-lightning-gpu/pull/75)
 
+* Fix data copy method in the state() method.
+[(#82)](https://github.com/PennyLaneAI/pennylane-lightning-gpu/pull/82)
```
Comment (AmintorDusko) [inline .github/CHANGELOG.md:71]:
```suggestion
* Fix data copy method in the state() method.
[(#82)](https://github.com/PennyLaneAI/pennylane-lightning-gpu/pull/82) 

```

**Review by AmintorDusko — DISMISSED** [review]
Looks good. Thank you for that!

File: pennylane_lightning_gpu/lightning_gpu.py [hunk 2]
```diff
@@ -229,7 +229,7 @@ def state(self):
             """Copy the state vector data from the device to the host. A state vector Numpy array is explicitly allocated on the host to store and return the data."""
             state = np.zeros(1 << self.num_wires, dtype=self.C_DTYPE)
             state = self._asarray(state, dtype=self.C_DTYPE)
-            self.syncH2D(state)
+            self.syncD2H(state)
```
Comment (mlxd) [inline pennylane_lightning_gpu/lightning_gpu.py:232]: Can we add a quick test to verify an arbitrary state can be copied back?
Even the dosctring could be a valid test.

**Review by mlxd — CHANGES_REQUESTED** [review]
thanks Shuli. One quick request

File: .github/CHANGELOG.md (hunk 1 above)
Comment (multiphaseCFD) [reply .github/CHANGELOG.md]: Good spot!

**Review by mlxd — APPROVED** [review]

**Review by AmintorDusko — APPROVED** [review]

File: pennylane_lightning_gpu/lightning_gpu.py [hunk 3]
```diff
@@ -226,10 +226,16 @@ def reset(self):
...
-            """Copy the state vector data from the device to the host. A state vector Numpy array is explicitly allocated on the host to store and return the data."""
+            """Copy the state vector data from the device to the host. A state vector Numpy array is explicitly allocated on the host to store and return the data.
+            **Example**
+            >>> dev = qml.device('lightning.gpu', wires=1)
+            >>> dev.apply([qml.PauliX(wires=[0])])
+            >>> print(dev.state)
+            [0.+0.j 1.+0.j]
+            """
             state = np.zeros(1 << self.num_wires, dtype=self.C_DTYPE)
             state = self._asarray(state, dtype=self.C_DTYPE)
```
Comment (chaeyeunpark) [inline pennylane_lightning_gpu/lightning_gpu.py:237]: Just one Q: Do we need `self._asarray` here? As Lightning-GPU device now inherits from `DefaultDevice`, this just calls `np.asarray` which I don't expect to do anything as `state` is already a numpy array.

**Pull request merged**